19) добавлен пакет react-router-dom

Chackmoole opened the pull request.

Chackmoole committed:
19) добавлен пакет react-router-dom

Chackmoole committed:
19)  Настроена маршрутизация в index. Из файла app вынесена разметка в UsersPage

Chackmoole committed:
19) Добавлены ссылки на маршрутизацию на главную страницу и страницу пользователей

Chackmoole committed:
19) Изменена структура страниц. Добавлен компоннт PageLayout. В него перенесены Header и Footer

Chackmoole committed:
19) Удалены лишние файлы стилей, так как теперь работают глобальные стили обертки.

Chackmoole committed:
19) Добавлены настройки tsconfig для импорта сразу через /pages.
Добалена настройка webpack для навигации по внутренним страницам

Chackmoole committed:
19) Добавлен дочерний роутинг в users, с отображением нового компонента CurrentUser.

NataliaIlina commented on package.json:
```diff
@@ -16,9 +16,12 @@
     "@mui/icons-material": "5.10.9",
     "@mui/material": "5.10.13",
+    "localforage": "^1.10.0",
+    "match-sorter": "^6.3.1",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "react-router-dom": "6.4.3"
+    "react-router-dom": "^6.4.3",
+    "sort-by": "^1.2.0"
```
много лишнего установил

NataliaIlina commented on src/components/Header/Header.tsx:
```diff
@@ -1,16 +1,19 @@
       <img src={logo} alt="react logo" width={40} height={40} />
-      <Typography variant="subtitle1">Страница пользователей</Typography>
+
+      <StyledNav>
+        <Link to="/">Главная страница</Link>
+        <Link to="/users">Страница пользователей</Link>
+      </StyledNav>
```
именно для меню больше подходит https://reactrouter.com/en/main/components/nav-link#navlink

там реализован индикатор текущей страницы

Chackmoole replied:
done

NataliaIlina commented on src/components/PageLayout/PageLayout.tsx:
```diff
@@ -0,0 +1,21 @@
+import { StyledPageLayout } from 'components/PageLayout/PageLayout.styled';
+
+interface IProps {
+  children?: React.ReactNode;
```
`children: React.ReactNode;`

по-хорошему это обязательный параметр

Chackmoole replied:
Была страница заглушка. done

Chackmoole committed:
19) Исправление по комментариям
Убраны лишние пакеты.
Параметр children в PageLayout - стал обязательным.
В Header Link заменен на более соответствующий для навигации NavLink.

Chackmoole committed:
19-20) реализована навигация по строкам на страницу пользователя по id.
Убран лишний console.log в Filters.

Chackmoole committed:
19-20) Убран лишний console.log в UserTables.

NataliaIlina commented on src/components/UsersTable/UsersTable.tsx:
```diff
@@ -42,6 +44,10 @@ const UsersTable = ({ openCreateModal, openEditModal, openDeleteModal }: IProps)
     );
   }, [currentStatus, isInclude, searchText]);
 
+  const onRowClick = (userId: number) => {
+    navigate(`${userId}`);
+  };
```
useCallback?

Chackmoole committed:
19-20) Добавлен useCallback соглано комментарию
